### PR TITLE
Bug 1796658 - fix notifications on failure of geckoview nightly bumps

### PR DESCRIPTION
**Pull request by jcristau**
A few years ago the owner and branch names for these bumps changed, so the notification would never actually happen.
Now we look for:
- the complete-pr task
- on the relbot/upgrade-geckoview-ac-main branch of the base repository

Since the branch name no longer includes the geckoview build id, we no longer include it in the email subject and replace it with the PR number.

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/main/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### GitHub Automation

Used by GitHub Actions.

### CODE REVIEW

File: taskcluster/ci/complete/kind.yml [hunk 1]
```diff
@@ -33,8 +33,8 @@ task-defaults:
     notifications:
         by-geckoview-bump:
             nightly:
-                subject: "[Android-Components] Failed to update to geckoview nightly {geckoview_timestamp}"
-                message: "Please check https://github.com/mozilla-mobile/android-components/pull/{pull_request_number}"
+                subject: "[Android-Components] Failed to update geckoview nightly PR#{pull_request_number}"
+                message: "Please check {repository}/pull/{pull_request_number}"
```
Comment (JohanLorenzo) [inline taskcluster/ci/complete/kind.yml:37]: Good idea on parameterizing that! 👍

File: taskcluster/ac_taskgraph/transforms/complete.py [hunk 2]
```diff
@@ -31,67 +26,63 @@ def resolve_keys(config, tasks):
...
-        task["name"] == "pr",  # Filter out pr-1 and pr-2
-        config.params["owner"] == "MickeyMoz@users.noreply.github.com",
-        config.params["head_repository"] == "https://github.com/MickeyMoz/android-components",
-        config.params["tasks_for"] == "github-pull-request",
-        _GECKOVIEW_NIGHTLY_BUMP_BRANCH_PATTERN.search(config.params["head_ref"]),
-    )):
+    if all(
+        (
+            task["name"] == "pr",  # Filter out pr-1 and pr-2
+            config.params["owner"] == "github-actions[bot]@users.noreply.github.com",
```
Comment (JohanLorenzo) [inline taskcluster/ac_taskgraph/transforms/complete.py:43]: Note to future readers: I confirm Github Actions are the only thing bumping geckoview today[1]

[1] https://phabricator.services.mozilla.com/D160665#5272763

File: taskcluster/ac_taskgraph/transforms/complete.py [hunk 3]
```diff
@@ -31,67 +26,63 @@ def resolve_keys(config, tasks):
...
+            config.params["head_repository"] == config.params["base_repository"],
```
Comment (JohanLorenzo) [inline taskcluster/ac_taskgraph/transforms/complete.py:44]: Note to future readers: Github Action creates branches on the base repo, not on a fork anymore.

File: taskcluster/ac_taskgraph/transforms/complete.py [hunk 4]
```diff
@@ -31,67 +26,63 @@ def resolve_keys(config, tasks):
...
+            config.params["tasks_for"] == "github-pull-request",
```
Comment (JohanLorenzo) [inline taskcluster/ac_taskgraph/transforms/complete.py:45]: Bug: with the recent change made in [bug 1797997](https://bugzilla.mozilla.org/show_bug.cgi?id=1797997) (#91), we should update the `tasks_for` here.

```suggestion
            config.params["tasks_for"] == "github-pull-request-untrusted",
```

As a matter of fact, the Github Actions user is considered untrusted (e.g.: https://github.com/mozilla-mobile/firefox-android/pull/123 and https://firefox-ci-tc.services.mozilla.com/tasks/RUmGD3w3QPaBuqSOKl4FmA)

**Review by JohanLorenzo — APPROVED** [review]
Looks good to me modulo a quick-to-fix bug 🙂

File: taskcluster/ac_taskgraph/transforms/complete.py (hunk 4 above)
Comment (jcristau) [reply taskcluster/ac_taskgraph/transforms/complete.py]: Thanks for catching that!

**Pull request merged**